File: python/성격 유형 검사하기.py
scores = {"R":0 , "T":0, "C":0, "F":0, 
              "J":0, "M":0, "A":0, "N":0}
    
    
def solution(survey, choices):
    answer = ''
    
    for idx, s_value in enumerate(survey):
        choice = choices[idx]
        getValue(s_value, choice)
    
    ks = list(scores.keys())
    vs = list(scores.values())
    
    for i, k in enumerate(ks):
        if i%2 == 0:
            if vs[i] < vs[i+1]:
                max_idx = i+1
            else:
                max_idx = i
            answer += ks[max_idx]
    
    return answer


def getValue(s_value, choice):
    if choice == 1:
        target = s_value[0]
        scores[target] += 3
    elif choice == 2:
        target = s_value[0]
        scores[target] += 2
    elif choice == 3:
        target = s_value[0]
        scores[target] += 1
    elif choice ==5:
        target = s_value[1]
        scores[target] += 1
    elif choice == 6:
        target = s_value[1]
        scores[target] += 2
    elif choice == 7:
        target = s_value[1]
        scores[target] += 3
        

File: java/성격유형검사하기.java
import java.util.HashMap;

class Solution {
    private HashMap<String, Integer> scores = 
        new HashMap<>(){{
            put("R", 0);
            put("T", 0);
            put("C", 0);
            put("F", 0);
            put("J", 0);
            put("M", 0);
            put("A", 0);
            put("N", 0);
        }};
    
    public String solution(String[] survey, int[] choices) {
        String answer = "";
        for(int i=0; i<survey.length; i++){
            String currentSurvey = survey[i];
            getValue(currentSurvey, choices[i]);
        }
        // System.out.println(scores);
        
        if(scores.get("R") >= scores.get("T")){
            answer += "R";
        }else{
            answer += "T";
        }
        
        if(scores.get("C") >= scores.get("F")){
            answer += "C";
        }else{
            answer += "F";
        }
        
        if(scores.get("J") >= scores.get("M")){
            answer += "J";
        }else{
            answer += "M";
        }
        
        if(scores.get("A") >= scores.get("N")){
            answer += "A";
        }else{
            answer += "N";
        }
        return answer;
    }
    
    private void getValue(String survey, int choice){
        if (choice==1){
            String target = String.valueOf(survey.charAt(0));
            int currentScore = scores.get(target);
            scores.put(target, currentScore + 3);
        }
        else if(choice == 2){
            String target = String.valueOf(survey.charAt(0));
            int currentScore = scores.get(target);
            scores.put(target, currentScore + 2);
        }else if(choice == 3){
            String target = String.valueOf(survey.charAt(0));
            int currentScore = scores.get(target);
            scores.put(target, currentScore + 1);
        }else if(choice ==5){
            String target = String.valueOf(survey.charAt(1));
            int currentScore = scores.get(target);
            scores.put(target, currentScore + 1);
        }else if(choice == 6){
            String target = String.valueOf(survey.charAt(1));
            int currentScore = scores.get(target);
            scores.put(target, currentScore + 2);
        }else if(choice == 7){
            String target = String.valueOf(survey.charAt(1));
            int currentScore = scores.get(target);
            scores.put(target, currentScore + 3);
        }
    }
}